src/SFA.DAS.Roatp.CourseManagement.Web.UnitTests/Controllers/AddAStandard/ConfirmNonRegulatedStandardControllerTests/ConfirmNonRegulatedStandardControllerGetTests.cs

```diff
@@ -0,0 +1,64 @@
+﻿using AutoFixture.NUnit3;
+using FluentAssertions;
+using MediatR;
+using Microsoft.AspNetCore.Mvc;
+using Moq;
+using NUnit.Framework;
+using SFA.DAS.Roatp.CourseManagement.Application.Standards.Queries.GetStandardInformation;
+using SFA.DAS.Roatp.CourseManagement.Web.Controllers.AddAStandard;
+using SFA.DAS.Roatp.CourseManagement.Web.Infrastructure;
+using SFA.DAS.Roatp.CourseManagement.Web.Models.AddAStandard;
+using SFA.DAS.Roatp.CourseManagement.Web.Services;
+using SFA.DAS.Roatp.CourseManagement.Web.UnitTests.TestHelpers;
+using SFA.DAS.Testing.AutoFixture;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace SFA.DAS.Roatp.CourseManagement.Web.UnitTests.Controllers.AddAStandard.ConfirmNonRegulatedStandardControllerTests
+{
+    [TestFixture]
+    public class ConfirmNonRegulatedStandardControllerGetTests
+    {
+        [Test, MoqAutoData]
+        public async Task Get_ModelMissingFromSession_RedirectsToStandardList(
+            [Frozen] Mock<IMediator> mediatorMock,
+            [Frozen] Mock<ISessionService> sessionServiceMock,
+            [Greedy] ConfirmNonRegulatedStandardController sut)
+        {
+            sut.AddDefaultContextWithUser();
+            sessionServiceMock.Setup(s => s.Get<StandardSessionModel>(It.IsAny<string>())).Returns((StandardSessionModel)null);
+
+            var response = await sut.GetConfirmationOfStandard();
+
+            var result = (RedirectToRouteResult) response;
+            result.Should().NotBeNull();
+            result.RouteName.Should().Be(RouteNames.ViewStandards);
+            mediatorMock.Verify(m => m.Send(It.IsAny<GetStandardInformationQuery>(), It.IsAny<CancellationToken>()), Times.Never);
+        }
+
```

Comment:
Needs a test for larscode=0 in the model

Reply:
Good spot. Added.